test/client_reset_test.dart

```diff
@@ -391,38 +409,38 @@ Future<void> main([List<String>? args]) async {
     await realm.syncSession.waitForUpload();
     await triggerClientReset(realm);
 
-    await waitFutureWithTimeout(onAfterCompleter.future, timeoutError: "onAfterReset is not reported.");
+    await onAfterCompleter.future.wait(defaultWaitTimeout, "onAfterReset is not reported.");
 
-    expect(onAfterResetOccurred, 1);
-    expect(onBeforeResetOccurred, 1);
+    expect(onAfterResetOccured, 1);
```

Comment:
```suggestion
    expect(onAfterResetOccurred, 1);
```
You are reintroducing a spelling error that was recently fixed here. I think you merged wrong.

I have created a branch with fixes

Reply:
I guess this is from a merge since this branch was based on older than `main` commit